test/e2e/framework.go

```diff
@@ -2504,6 +2514,22 @@ func (data *TestData) copyNodeFiles(nodeName string, fileName string, covDir str
 	return nil
 }
 
+// createPodWithVolume creates a new Pod in the test namespace and mount a ConfigMap through a Volume inside the Pod.
+func (data *TestData) createPodWithVolume(name string, ns string, ctrName string, hostNetwork bool, cm corev1.ConfigMap) error {
+	volumeMount := corev1.VolumeMount{Name: "config-volume", MountPath: "/etc/config/"}
+	volume := corev1.Volume{
+		Name: "config-volume",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: cm.Name,
+				},
+			},
+		},
+	}
+	return NewPodBuilder(name, ns, agnhostImage).WithCommand([]string{"sleep", "3600"}).WithContainerName(ctrName).MountConfigMap([]corev1.VolumeMount{volumeMount}, []corev1.Volume{volume}).WithHostNetwork(hostNetwork).Create(data)
+}
```

Comment:
the point of `PodBuilder` is that we avoid creating _new_ highly specialized Pod creation functions like this one in framework.go